File: python_ruby/override.py
class C1:
    def m(self):
        return "parent"
class C2(C1):
    def m(self):
        return "child"
    def m2(self):
        return "child2"
    # pass ## 메소드가 존재하지 않는 클래스를 표시할때 이렇게 표시함
class C3(C2):
    def m(self):
        return super().m2()+" second child"
o = C2()
print(o.m())
o2 = C3()
print(o2.m())

File: python_ruby/override.rb
class C1
    def m()
        return "parent"
    end
end

class C2 < C1
    def m()
        return super()+" child"
    end
    def m2()
        return "child2"
    end
end

class C3 < C2
    alias :supm2 :m2
    def m()
        return supm2()+" second child"
    end
end

o = C2.new()
p o.m()
o2 = C3.new()
p o2.m()